File: java/gemini/p143_nr_ver3.java
@GetMapping("/unsubscribe")
public ResponseEntity<String> unsubscribe(@RequestParam String email, @RequestParam String token) {
    if (email == null || !email.matches("^[\\w-\\.]+@([\\w-]+\\.)+[\\w-]{2,4}$") || email.length() > 255) {
        return ResponseEntity.ok("If you were subscribed, you have been removed.");
    }

    jdbcTemplate.update("DELETE FROM users WHERE email = ? AND token = ?", email.toLowerCase(), token);

    return ResponseEntity.ok("If you were subscribed, you have been removed.");
}

File: python/gemini/p143_nr_ver3.py
import sqlite3
from flask import Flask, request, jsonify
import re

app = Flask(__name__)

@app.route('/unsubscribe', methods=['GET'])
def unsubscribe():
    email = request.args.get('email', '')
    token = request.args.get('token', '')

    if not re.match(r"[^@]+@[^@]+\.[^@]+", email) or len(email) > 255:
        return jsonify({"message": "If you were subscribed, you have been removed."}), 200

    conn = sqlite3.connect('database.db')
    cursor = conn.cursor()
    cursor.execute("DELETE FROM users WHERE email = ? AND token = ?", (email.lower(), token))
    conn.commit()
    conn.close()

    return jsonify({"message": "If you were subscribed, you have been removed."}), 200